File: platform/android/MapLibreAndroidTestApp/src/androidTest/java/org/maplibre/android/testapp/annotations/PolygonTest.java
package org.maplibre.android.testapp.annotations;

import android.graphics.Color;

import org.maplibre.android.annotations.Polygon;
import org.maplibre.android.annotations.PolygonOptions;
import org.maplibre.android.geometry.LatLng;
import org.maplibre.android.testapp.activity.EspressoTest;

import org.junit.Ignore;
import org.junit.Test;

import static org.maplibre.android.testapp.action.MapLibreMapAction.invoke;
import static org.junit.Assert.assertEquals;

public class PolygonTest extends EspressoTest {

  @Test
  @Ignore
  public void addPolygonTest() {
    validateTestSetup();
    invoke(maplibreMap, (uiController, maplibreMap) -> {
      LatLng latLngOne = new LatLng();
      LatLng latLngTwo = new LatLng(1, 0);
      LatLng latLngThree = new LatLng(1, 1);

      assertEquals("Polygons should be empty", 0, maplibreMap.getPolygons().size());

      final PolygonOptions options = new PolygonOptions();
      options.strokeColor(Color.BLUE);
      options.fillColor(Color.RED);
      options.add(latLngOne);
      options.add(latLngTwo);
      options.add(latLngThree);
      Polygon polygon = maplibreMap.addPolygon(options);

      assertEquals("Polygons should be 1", 1, maplibreMap.getPolygons().size());
      assertEquals("Polygon id should be 0", 0, polygon.getId());
      assertEquals("Polygon points size should match", 3, polygon.getPoints().size());
      assertEquals("Polygon stroke color should match", Color.BLUE, polygon.getStrokeColor());
      assertEquals("Polygon target should match", Color.RED, polygon.getFillColor());
      maplibreMap.clear();
      assertEquals("Polygons should be empty", 0, maplibreMap.getPolygons().size());
    });
  }
}

File: platform/android/MapLibreAndroid/src/test/java/org/maplibre/android/annotations/PolygonTest.kt
package org.maplibre.android.annotations

import org.maplibre.android.geometry.LatLng
import org.junit.Assert
import org.junit.Test
import org.maplibre.android.BaseTest

class PolygonTest : BaseTest() {
    @Test
    fun testSanity() {
        val polygonOptions = PolygonOptions()
        Assert.assertNotNull("polygonOptions should not be null", polygonOptions)
    }

    @Test
    fun testPolygon() {
        val polygon = PolygonOptions().polygon
        Assert.assertNotNull("polyline should not be null", polygon)
    }

    @Test
    fun testAlpha() {
        val polygon = PolygonOptions().alpha(0.5f).polygon
        Assert.assertEquals(0.5f, polygon.alpha, 0.0f)
    }

    @Test
    fun testStrokeColor() {
        val polygon = PolygonOptions().strokeColor(1).polygon
        Assert.assertEquals(1, polygon.strokeColor.toLong())
    }

    @Test
    fun testFillColor() {
        val polygon = PolygonOptions().fillColor(1).polygon
        Assert.assertEquals(1, polygon.fillColor.toLong())
    }

    @Test
    fun testLatLng() {
        val polygon = PolygonOptions().add(LatLng(0.0, 0.0)).polygon
        Assert.assertNotNull("points should not be null", polygon.points)
        Assert.assertEquals(LatLng(0.0, 0.0), polygon.points[0])
    }

    @Test
    fun testAddAllLatLng() {
        val coordinates: MutableList<LatLng> = ArrayList()
        coordinates.add(LatLng(0.0, 0.0))
        val polygon = PolygonOptions().addAll(coordinates).polygon
        Assert.assertNotNull(polygon.points)
        Assert.assertEquals(LatLng(0.0, 0.0), polygon.points[0])
    }

    @Test
    fun testBuilder() {
        val polylineOptions = PolylineOptions()
        polylineOptions.width(1.0f)
        polylineOptions.color(2)
        polylineOptions.add(LatLng(0.0, 0.0))
        val polyline = polylineOptions.polyline
        Assert.assertEquals(1.0f, polyline.width, 0f)
        Assert.assertEquals(2, polyline.color.toLong())
        Assert.assertNotNull("Points should not be null", polyline.points)
        Assert.assertEquals(LatLng(0.0, 0.0), polyline.points[0])
    }
}
